File: src/main/java/com/jcloisterzone/io/message/AiMessage.java
package com.jcloisterzone.io.message;

import com.jcloisterzone.io.MessageCommand;

@MessageCommand("AI")
public class AiMessage extends AbstractMessage implements ReplayableMessage, RandomChangingMessage {

    private Double random;
    
    private Integer player;

    private Integer seq;

    public AiMessage() {
    }

    @Override
    public Double getRandom() {
        return random;
    }

    @Override
    public void setRandom(Double random) {
        this.random = random;
    }

    public Integer getPlayer() {
        return player;
    }

    public void setPlayer(Integer player) {
        this.player = player;
    }

    public Integer getSeq() {
        return seq;
    }

    public void setSeq(Integer seq) {
        this.seq = seq;
    }
}

File: src/main/ts/com/jcloisterzone/io/message/AiMessage.ts
import { AbstractMessage } from "./AbstractMessage.js";
import type { RandomChangingMessage } from "./RandomChangingMessage.js";
import type { ReplayableMessage } from "./ReplayableMessage.js";

/** AI — request the engine to compute the move for the given (active) player. The
 *  engine replies with an AI_MESSAGE wrapping the chosen move. */
export class AiMessage extends AbstractMessage implements ReplayableMessage, RandomChangingMessage {
  static readonly command = "AI";

  private random: number | null = null;
  private player: number | null = null;
  private seq: number | null = null;

  getRandom(): number | null {
    return this.random;
  }
  setRandom(random: number | null): void {
    this.random = random;
  }
  getPlayer(): number | null {
    return this.player;
  }
  setPlayer(player: number | null): void {
    this.player = player;
  }
  getSeq(): number | null {
    return this.seq;
  }
  setSeq(seq: number | null): void {
    this.seq = seq;
  }
}
